src/functions.js

```diff
@@ -70,26 +70,48 @@ defineFunction("mclass", [
 defineFunction("mclass", ["\\stackrel", "\\overset", "\\underset"], {
     numArgs: 2,
 }, function(context, args) {
-    const mathAxisArg = args[1];
+    const baseArg = args[1];
     const shiftedArg = args[0];
 
-    const xAxisOp = new ParseNode("op", {
+    let mclass = "mrel";  // default. May change below.
+    if (context.funcName !== "\\stackrel") {
+        // LaTeX applies \binrel spacing to \overset and \underset.
+        // \binrel spacing varies with (bin|rel|ord) of the atom in the argument.
+        // We'll do the same.
+        let atomType = "";
+        if (baseArg.type === "ordgroup") {
+            atomType = baseArg.value[0].type;
+        } else {
+            atomType = baseArg.type;
+        }
+        if (/^(bin|rel)$/.test(atomType)) {
+            mclass = "m" + atomType;
+        } else {
+            // This may capture some instances in which the baseArg is more than
+            // just a single symbol. Say a \overset inside an \overset.
+            // TODO: A more comprehensive way to determine the baseArg type.
+            mclass = "mord";
+        }
+    }
+
+    const baseOp = new ParseNode("op", {
         type: "op",
         limits: true,
         alwaysHandleSupSub: true,
         symbol: false,
-        value: ordargument(mathAxisArg),
-    }, mathAxisArg.mode);
+        suppressBaseShift: context.funcName !== "\\stackrel",
+        value: ordargument(baseArg),
+    }, baseArg.mode);
```

Comment:
`ParseNode`s are flow-typed by #1276:
https://github.com/Khan/KaTeX/blob/5a4aedd8827776acd737c993b9416c36302465f7/src/ParseNode.js#L66-L73

Reply:
@ylemkimon Thank you for adding `suppressBaseShift` to the flow type enumeration. I was trying to think of a way to make `\underset` work without doing that. I was getting nowhere.